jmclient/jmclient/maker.py

```diff
@@ -139,14 +139,30 @@ def on_tx_received(self, nick, txhex, offerinfo):
             our_inputs[index] = (script, amount)
 
         txs = self.wallet.sign_tx(btc.deserialize(unhexlify(txhex)), our_inputs)
-
         for index in our_inputs:
-            sigmsg = txs['ins'][index]['script']
+            sigmsg = unhexlify(txs['ins'][index]['script'])
```

Comment:
`sign_tx` is called with non-hex data and the return data should be non-hex, too. If it's not that seems wrong.

Reply:
It's not quite like that. I'm going to add a commit updating the comments to btc.serialize in particular, and also btc.sign, because the former especially describes what happens wrong.

serialize() is using *any* outpoint:hash field being hex in the input transaction dict as a flag to output hex serialization. sign() just obeys the encoding of the input *serialized* transaction; it accepts only a serialized tx, and outputs a hex re-serialization if input is hex, and binary if binary.

Yes this is a slight delta to previous (in particular the sign()) behaviour. I know we obviously we want to completely change this, but for now at least I'm going to try to make sure the comments are accurate.

Reply:
Thanks for paying careful attention to this; in fact the comment was *still* wrong; I have updated in  . In short, jmbitcoin.sign() *always* returns hex as currently coded, hence wallet.sign_tx does also.

So overall: deserialize() respects encoding of input; serialize chooses hex if any hash input is hex; sign always returns hex (and only works with serialized form, cf other comments).